packages/data-seeder/src/locations.ts

```diff
@@ -227,31 +227,48 @@ export async function seedLocations(token: string) {
   const locations = (await getLocations()).filter((location) => {
     return !savedLocationsSet.has(location.id)
   })
-  const res = await fetch(`${GATEWAY_HOST}/location?`, {
-    method: 'POST',
-    headers: {
-      Authorization: `Bearer ${token}`,
-      'Content-Type': 'application/fhir+json'
-    },
-    body: JSON.stringify(
-      locations
-        // statisticalID & code are legacy properties
-        .map(({ id, locationType, ...loc }) => ({
-          statisticalID: id,
-          code: locationType,
-          ...loc
-        }))
+  const locationSegments: TypeOf<typeof LocationSchema>[] = []
+  jurisdictionLevels.forEach((jurisictionType) => {
+    const jurisictionTypeLocations = locations.filter(
+      (location) =>
+        location.jurisdictionType === jurisictionType &&
+        location.locationType === 'ADMIN_STRUCTURE'
     )
+    if (jurisictionTypeLocations.length) {
+      locationSegments.push(jurisictionTypeLocations)
+    }
   })
-  if (!res.ok) {
-    raise(await res.json())
-  }
-  const response: fhir3.Bundle<fhir3.BundleEntryResponse> = await res.json()
-  response.entry?.forEach((res, index) => {
-    if (res.response?.status !== '201') {
-      console.log(
-        `Failed to create location resource for: "${locations[index].name}"`
+  // offices, health faclities
+  locationSegments.push(
+    locations.filter((location) => location.locationType !== 'ADMIN_STRUCTURE')
+  )
+  for (const segment of locationSegments) {
+    const res = await fetch(`${GATEWAY_HOST}/location?`, {
+      method: 'POST',
+      headers: {
+        Authorization: `Bearer ${token}`,
+        'Content-Type': 'application/fhir+json'
+      },
+      body: JSON.stringify(
+        segment
+          // statisticalID & code are legacy properties
+          .map(({ id, locationType, ...loc }) => ({
+            statisticalID: id,
+            code: locationType,
+            ...loc
+          }))
       )
+    })
+    if (!res.ok) {
+      raise(await res.json())
     }
-  })
+    const response: fhir3.Bundle<fhir3.BundleEntryResponse> = await res.json()
```

Comment:
@tahmidrahman-dsi The issue might be that we need to resolve the partOf values of the district locations with that of the actual values we get from hearth after creating the states.

Reply:
the these level by level segmentation might need to be moved to the actual handler in gateway due to this reason